radio/src/os/task.h

```diff
@@ -43,14 +43,74 @@ bool mutex_lock(mutex_handle_t* h);
 void mutex_unlock(mutex_handle_t* h);
 bool mutex_trylock(mutex_handle_t* h);
 
+
+// this helper class is only to be used on the stack and never to be shared outside the scope where it was created
+/* example usage
+ *
+ * void doWork()
```

Comment:
```c++
void doWork()
{
  while (true) {
    MutexLock lock = MutexLock::MakeInstance(mutex);
    if (queue.isEmpty()) {
      break;
    }
    auto item = queue.pop(); // better to protect the pop() in the same shot
    //   <do something with item>
  } // auto mutex unlock, because lock is destructed here
}